sklearn/utils/_testing.py

```diff
@@ -655,14 +655,25 @@ def check_docstring_parameters(func, doc=None, ignore=None):
 
     # Analyze function's docstring
     if doc is None:
-        with warnings.catch_warnings(record=True) as w:
+        records = []
+        with warnings.catch_warnings(record=True):
+            warnings.simplefilter("error", UserWarning)
             try:
                 doc = docscrape.FunctionDoc(func)
+            except UserWarning as exp:
+                if "potentially wrong underline length" in str(exp):
+                    # Catch warning raised as of numpydoc 1.2 when
+                    # the underline length for a section of a docstring
+                    # is not consistent.
+                    message = str(exp).split("\n")[:4]
+                    incorrect += [f"In function: {func_name}"] + message
```

Comment:
This is fishy in particular. message is an `str` while `incorrect` is a `list`. Did you intend the following?

```suggestion
                    incorrect += [f"In function {func_name}: {message}"]
```

Reply:
In this case, `message` is a list because of the `split("\n")`. The intention was to convert the UserWarning message with newlines into a list.

Reply:
ah ok :)